sauna/plugins/ext/network.py

```diff
@@ -0,0 +1,130 @@
+        super().__init__(config)
+
+    @my_plugin.check()
+    def upload_kbytes_speed(self, check_config):
+        ul, _, _, _ = self.get_network_data(
+            interface=check_config['interface'])
+        ul = round(ul, 2)
+        return (
+            self._value_to_status_less(ul, check_config),
+            'Upload speed: {} KB/s'.format(ul)
+        )
+
+    @my_plugin.check()
+    def download_kbytes_speed(self, check_config):
+        _, dl, _, _ = self.get_network_data(
+            interface=check_config['interface'])
+        dl = round(dl, 2)
+        return (
+            self._value_to_status_less(dl, check_config),
+            'Download speed: {} KB/s'.format(dl)
+        )
+
+    @my_plugin.check()
+    def cumulative_kbytes_speed(self, check_config):
+        ul, dl, _, _ = self.get_network_data(
+            interface=check_config['interface'])
+        total = ul + dl
+        total = round(total, 2)
+        return (
+            self._value_to_status_less(total, check_config),
+            'Total speed: {} KB/s'.format(total)
+        )
+
+    @my_plugin.check()
+    def upload_packet_speed(self, check_config):
+        _, _, ul, _ = self.get_network_data(
+            interface=check_config['interface'])
+        ul = round(ul, 2)
+        return (
+            self._value_to_status_less(ul, check_config),
+            'Upload : {} p/s'.format(ul)
+        )
+
+    @my_plugin.check()
+    def download_packet_speed(self, check_config):
+        _, _, _, dl = self.get_network_data(
+            interface=check_config['interface'])
+        dl = round(dl, 2)
+        return (
+            self._value_to_status_less(dl, check_config),
+            'Download : {} p/s'.format(dl)
+        )
+
+    @my_plugin.check()
+    def cumulative_packet_speed(self, check_config):
+        _, _, ul, dl = self.get_network_data(
+            interface=check_config['interface'])
+        total = ul + dl
+        total = round(total, 2)
+        return (
+            self._value_to_status_less(total, check_config),
+            'Total : {} p/s'.format(total)
+        )
+
+    def get_network_data(self, interface='eth0', delay=1):
+        t0 = time.time()
+        counter = self.psutil.net_io_counters(pernic=True)[interface]
+        first_values = (counter.bytes_sent, counter.bytes_recv,
+                        counter.packets_sent, counter.packets_recv)
+
+        time.sleep(delay)
+        counter = self.psutil.net_io_counters(pernic=True)[interface]
+        t1 = time.time()
+        last_values = (counter.bytes_sent, counter.bytes_recv,
+                       counter.packets_sent, counter.packets_recv)
+        kb_ul, kb_dl, p_ul, p_dl = [
+            (last - first) / (t1 - t0)
+            for last, first in zip(last_values, first_values)
+        ]
+        return kb_ul / 1000.0, kb_dl / 1000.0, p_ul, p_dl
+
+    @staticmethod
+    def config_sample():
+        return '''
+  - type: Network
+
+    checks:
```

Comment:
It's not entirely clear at the first glance if the alert will be raised if the speed is lower or higher than the threshold.